src/Chamilo/UserBundle/Repository/UserRepository.php

```diff
@@ -431,4 +436,99 @@ public function getAssignedHrmUserList($userId, $urlId)
+     *
+     * @return array $values
+     */
+    public function getPersonalDataToJson($userId, array $substitutionTerms)
+    {
+        /** @var User $user */
+        $user = $this->find($userId);
+
+        $user->setPassword($substitutionTerms['password']);
+        $user->setSalt($substitutionTerms['salt']);
+        $noDataLabel = $substitutionTerms['empty'];
+
+        // Dummy content
+        $user->setDateOfBirth(null);
+        $user->setBiography($noDataLabel);
+        $user->setFacebookData($noDataLabel);
+        $user->setFacebookName($noDataLabel);
+        $user->setFacebookUid($noDataLabel);
+        $user->setImageName($noDataLabel);
+        $user->setTwoStepVerificationCode($noDataLabel);
+        $user->setGender($noDataLabel);
+        $user->setGplusData($noDataLabel);
+        $user->setGplusName($noDataLabel);
+        $user->setGplusUid($noDataLabel);
+        $user->setLocale($noDataLabel);
+        $user->setTimezone($noDataLabel);
+        $user->setTwitterData($noDataLabel);
+        $user->setTwitterName($noDataLabel);
+        $user->setTwitterUid($noDataLabel);
+        $user->setWebsite($noDataLabel);
+        $user->setToken($noDataLabel);
+        $user->setCourses([]);
+        $user->setClasses([]);
+        $user->setDropBoxSentFiles([]);
+        $user->setDropBoxReceivedFiles([]);
+        $user->setGroups([]);
+        $user->setCurriculumItems([]);
+        $user->setPortals([]);
+        $user->setExtraFields([]);
+        $user->setSessionCourseSubscriptions([]);
+        $user->setSessionAsGeneralCoach([]);
+
+        $lastLogin = $user->getLastLogin();
+        if (empty($lastLogin)) {
+            $login = $this->getLastLogin($user);
+            $lastLogin = $login->getLoginDate();
+
+        }
+        $user->setLastLogin($lastLogin);
+
+        $dateNormalizer = new GetSetMethodNormalizer();
+
+        $callback = function ($dateTime) {
+            return $dateTime instanceof \DateTime
+                ? $dateTime->format(\DateTime::ISO8601)
+                : '';
+        };
+        $dateNormalizer->setCallbacks(['createdAt' => $callback, 'lastLogin' => $callback]);
+
+        $normalizers = [$dateNormalizer];
+        $serializer = new Serializer($normalizers, [new JsonEncoder()]);
+
+        $jsonContent = $serializer->serialize($user, 'json');
+
+        return $jsonContent;
+    }
+
+    /**
+     * @param User $user
+     *
+     * @return null|TrackELogin
+     */
+    public function getLastLogin(User $user)
+    {
+        $repo = $this->getEntityManager()->getRepository('ChamiloCoreBundle:TrackELogin');
+        $qb = $repo->createQueryBuilder('l');
+
+        $login = $qb
+            ->select('l')
+            ->where(
+                $qb->expr()->eq('l.loginUserId', $user->getId())
+            )
+            ->setMaxResults(1)
+            ->orderBy('l.loginDate', 'DESC')
+            ->getQuery()
+            ->getOneOrNullResult();
+
+        return $login;
+    }
```

Comment:
I see what you did there :wink: 
Now I see how to avoid adding the ORM association

Reply:
Yes :)